bokehjs/src/coffee/util/zoom.coffee

```diff
@@ -0,0 +1,80 @@
+scale_range = ({frame, factor, center, h_axis_only = false, v_axis_only = false}) ->
```

Comment:
@birdsarah, @bryevdv , @mattpap  and anyone else who loves the tests...

I've been learning up on Mocha and sinon and looking at the tests that exist for things like CartesianFrame and Canvas.  It seems like since the current implementation of `scale_range` depends on getting calculated properties like `h_range` and `v_range` and the `mappers`, testing this code is going to be challenging.

1. Are there any JS tests that actually use a document with the solver and everything to test things like this?  Would we need something like CasperJS/PhosphorJS in order to do that?  Would it be worth adding those types of tests?
1. Would it be better to stub CartesianFrame to not require that the  
1. Should I refactor `scale_range` so that it is a pure helper and doesn't depend on having a "solved" CartesianFrame?  In other words, should I pass `h_range`, `v_range`, `x_mappers` and `y_mappers` instead of passing in `frame`?